earth_enterprise/src/fusion/dbmanifest/dbmanifest.cpp

```diff
@@ -327,6 +332,63 @@ void DbManifest::GetManifest(geFilePool &file_pool,
   GetXmlManifest(stream_manifest);
 }
 
+// Read POI file and extract the data file names from within it and then add
+// them to the manifest entries as dependant files
+void DbManifest::GetPoiDataFiles(ManifestEntry* stream_manifest_entry,
+                                 ManifestEntry* search_manifest_entry)
+{
+  const std::string& poi_file = search_manifest_entry->current_path;
+  assert(stream_manifest_entry->current_path == search_manifest_entry->current_path);
+  notify(NFY_DEBUG,
+        "Parsing POI file %s looking for data files", poi_file.c_str());
+  khxml::DOMLSParser *parser = CreateDOMParser();
+  if (parser) {
+    khxml::DOMDocument *doc = ReadDocument(parser, poi_file);
+    if (doc) {
+      try {
+        if (khxml::DOMElement *root = doc->getDocumentElement()) {
+          if (khxml::DOMElement* el_search_table = GetFirstNamedChild(root, "SearchTableValues")) {
+            std::vector<std::string> data_files;
+            FromElementWithChildName(el_search_table, "SearchDataFile", data_files);
+            if (data_files.empty()) {
+              notify(NFY_WARN,
+                    "No data files defined for POI file %s", poi_file.c_str());
+            }
+            for (uint i = 0; i < data_files.size(); ++i) {
+
+              const std::string& data_file = data_files[i];
+              
+              // poi data files have to be set aside until after AddDB processing is done
+              // so we hide them in a special location now within the manifest and will
+              // use them when we crate an 'upload' manifest during DBsync processing
+              if (stream_manifest_entry != search_manifest_entry) {
```

Comment:
I'm trying really hard to determine if this line is doing the right thing. I think the behavior is a little different than the previous version. `stream_manifest != search_manifest` will return false if the vectors are different but have identical contents.  Since this version is comparing pointers, it will return true for different vectors with identical contents. Of course, it's possible that there was a bug in the original version and this is better. The safest path is probably to do the comparison outside this function and pass a boolean in to indicate whether this line should be executed.

Reply:
These are pointers the line is not comparing that the values are the but that the instances are different.  Which is what we had before and what we still want.  If the instances are different we want to be sure to add the same data to both instances.  If they are the same then we want to to be sure to not to double add the data.  If we had shared_ptr<>'s we could optimize the code to have two manifests where one happened to point to a sub-set of of the others entries.  Doing that with raw pointers gets ugly....  This isn't the most space efferent solution but that was how the original code worked.

Reply:
Would we be able to enable the c++11 flag on a per-executable or per CPP basis? I don't like the idea of mixing the project with some c++98 and some c++11 files, but it may be the only practical way for us to start upgrading.